### 문제

#### 1. 편두통에 대한 치료효과를 비교실험한 자료로서 새 치료약과 위약을 각각 투여한 두 집단에 대해 진료병원별로 조사한 결과이다. 각 병원의 환자들에게 실제 약과 위약 중 어느 한 가지를 랜덤하게 선택하도록 하였다.

|||성공|실패|
|:-:|:-:|:-:|:-:|
|병원1|집단_약|6|4|
|병원1|집단_위약|2|8|
|병원2|집단_약|4|3|
|병원2|집단_위약|1|5|
|병원3|집단_약|5|3|
|병원3|집단_위약|3|6|

#### (1) 부분분할표의 조건부 오즈비를 구하여라. 이 값이 병원들간에 유사한가?

<병원 1의 경우 오즈비>
<img src="https://latex.codecogs.com/svg.image?&space;Odds1=\frac{6/4}{2/8}=\frac{6\cdot8}{2\cdot4}=\frac{48}{8}=6.0">

<병원 2의 경우 오즈비>
<img src="https://latex.codecogs.com/svg.image?&space;Odds2=\frac{4/3}{1/5}=\frac{4\cdot5}{1\cdot3}=\frac{20}{3}=6.67">

<병원 3의 경우 오즈비>
<img src="https://latex.codecogs.com/svg.image?&space;Odds3=\frac{5/3}{3/6}=\frac{5\cdot6}{3\cdot3}=\frac{30}{9}=3.33">

병원1과 병원2의 오즈비는 각각 6.0과 6.67로 유사하지만, 병원 3의 오즈비는 3.33으로 다른 두 병원의 오즈비에 비해 작다고 볼 수 있다.

#### (2) 집단과 반응간의 조건부독립성 검정에 대한 P-값을 구하여라.  
   ==> 코크란-맨틀-핸첼 검정 이용

In [1]:
# 데이터 입력
data <- array(c(6, 4, 2, 8,
                4, 3, 1, 5,
                5, 3, 3, 6),
              dim = c(2, 2, 3))

# 코크란-맨틀-핸첼 검정
result <- mantelhaen.test(data)

# 결과 출력
print(result)


	Mantel-Haenszel chi-squared test with continuity correction

data:  data
Mantel-Haenszel X-squared = 5.0879, df = 1, p-value = 0.02409
alternative hypothesis: true common odds ratio is not equal to 1
95 percent confidence interval:
  1.43094 16.88727
sample estimates:
common odds ratio 
         4.915757 



코크란-맨텔-핸첼 검정 결과 p값은 0.02409이다.  



#### (3) 집단과 반응간의 평균 조건부연관성을 추정하고 해석하라.

Mantel-Haenszel 함수 실행 결과에 따르면 공통 오즈비는 4.916이다.  
이것은 약을 복용한 집단의 오즈비가 위약을 복용한 집단의 오즈비보다 4.916배 크다는 의미이다.  

95% 유의수준에서 오즈비 신뢰구간은 [1.431 ~ 16.887]이며, 구간에 1이 포함되어 있지 않기 때문에 귀무가설을 기각할 수 있다.  
약을 먹은 사람의 오즈비가 위약을 먹은 사람의 오즈비에 비해 더 크다고 할 수 있다.

#### (4) 세 병원의 오즈비가 모두 같은지를 검정하여라.

세 병원의 오즈비가 같은지 검정하기 위해 브레슬로-데이 검정을 활용한다.

In [4]:
library(DescTools)

# Breslow-Day 검정 수행 (오즈비의 동질성 검정)
breslow_day_result <- BreslowDayTest(data)

# 결과 출력
print(breslow_day_result)


	Breslow-Day test on Homogeneity of Odds Ratios

data:  data
X-squared = 0.23691, df = 2, p-value = 0.8883



검정결과 p-값은 0.888로 매우 크다. p-값이 세 병원의 오즈비가 동일하다는 귀무가설을 기각하지 못한다.  
즉, 세 병원의 오즈비는 같다고 할 수 있다.